### PR TITLE
[MBL-15923]  - Introduce graphql to dataseeding test framework & Implement Comment Library E2E test

### PR DESCRIPTION
**Pull request by kdeakinstructure**
Introduce GraphQL to dataseeding for tests and implement comment library e2e test case.

refs: MBL-15923
affects: Teacher
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/teacher/src/androidTest/java/com/instructure/teacher/ui/e2e/CommentLibraryE2ETest.kt [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+        val request = UserSettingsApiModel(
+            manualMarkAsRead = false,
+            collapseGlobalNav = false,
+            hideDashCardColorOverlays = false,
+            commentLibrarySuggestions = true
+        )
+        UserApi.putSelfSettings(teacher.id, request) // Set comment library "Show suggestions when typing" user settings to be able to see the library comments.
+
+        val apolloClient = ApolloClient.builder()
```
Comment (tamaskozmer) [inline apps/teacher/src/androidTest/java/com/instructure/teacher/ui/e2e/CommentLibraryE2ETest.kt:90]: I think all this code from here until the `tokenLogin` belongs to the dataseeding library. I would move this `apolloClient` to the `CanvasRestAdapter` where all the other network handling properties are held, and for the network calls I would create an object in the `api` package. The code here should look like something like this:

```
CommentLibraryApi.createComment(course.id, "Test Comment")
CommentLibraryApi.createComment(course.id, "Test Comment 2")
```

File: apps/teacher/src/androidTest/java/com/instructure/teacher/ui/e2e/CommentLibraryE2ETest.kt [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+class CommentLibraryE2ETest : TeacherTest() {
+
+    override fun displaysPageObjects() {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    @E2E
+    @Test
+    @TestMetaData(Priority.P0, FeatureCategory.DASHBOARD, TestCategory.E2E)
+    fun testCommentLibraryE2E() {
```
Comment (tamaskozmer) [inline apps/teacher/src/androidTest/java/com/instructure/teacher/ui/e2e/CommentLibraryE2ETest.kt:56]: There is a lot of dataseeding before we start the test, I would move this to a separate function, to make the actual test steps more clear.

File: apps/teacher/src/androidTest/java/com/instructure/teacher/ui/pages/SpeedGraderPage.kt [hunk 3]
```diff
@@ -79,7 +79,7 @@ class SpeedGraderPage : BasePage() {
     }
 
     fun selectFilesTab(fileCount: Int) {
-        val filesTab = waitForViewWithText(getStringFromResource(R.string.sg_tab_files_w_counter, fileCount).uppercase(Locale.getDefault()))
+        val filesTab = waitForViewWithText(getStringFromResource(R.string.sg_tab_files_w_counter, fileCount).toUpperCase())
```
Comment (tamaskozmer) [inline apps/teacher/src/androidTest/java/com/instructure/teacher/ui/pages/SpeedGraderPage.kt:82]: What's the reason behind this change? I changed this in MBL-15645, because `toUpperCase()` is deprecated end eventually will be removed in a newer Kotlin version.

File: automation/dataseedingapi/build.gradle [hunk 4]
```diff
@@ -11,13 +12,23 @@ buildscript {
 
     dependencies {
         classpath Plugins.KOTLIN
+        classpath 'com.apollographql.apollo:apollo-gradle-plugin:2.5.9'
```
Comment (tamaskozmer) [inline automation/dataseedingapi/build.gradle:15]: We can use the dependency that is defined in the `GlobalDependencies` file.

File: automation/dataseedingapi/build.gradle [hunk 5]
```diff
@@ -34,6 +45,9 @@ dependencies {
 
     implementation 'org.jsoup:jsoup:1.11.3'
     implementation 'com.github.javafaker:javafaker:0.17.2'
+    implementation 'com.apollographql.apollo:apollo-runtime:2.5.9'
```
Comment (tamaskozmer) [inline automation/dataseedingapi/build.gradle:48]: We can use the dependency that is defined in the `GlobalDependencies` file.

File: automation/dataseedingapi/build.gradle [hunk 6]
```diff
@@ -34,6 +45,9 @@ dependencies {
...
+    implementation 'com.apollographql.apollo:apollo-coroutines-support:2.5.9'
```
Comment (tamaskozmer) [inline automation/dataseedingapi/build.gradle:49]: I think we don't need the coroutines support lib.

File: automation/dataseedingapi/src/main/kotlin/com/instructure/dataseeding/api/UserApi.kt [hunk 7]
```diff
@@ -48,13 +50,28 @@ object UserApi {
...
+        fun putSelfSettings(@Path("userId") userId: Long, @Body body: UserSettingsApiModel): Call<UserSettingsApiModel>
+
     }
 
     private val userAdminService: UserService by lazy {
         CanvasRestAdapter.adminRetrofit.create(UserService::class.java)
     }
 
-    fun createCanvasUser(
+    fun getSelfSettings(userId: Long): UserSettingsApiModel {
```
Comment (tamaskozmer) [inline automation/dataseedingapi/src/main/kotlin/com/instructure/dataseeding/api/UserApi.kt:65]: This and the corresponding interface function are never used.

File: automation/dataseedingapi/src/main/kotlin/com/instructure/dataseeding/util/CanvasRestAdapter.kt [hunk 8]
```diff
@@ -69,6 +69,16 @@ object CanvasRestAdapter {
                 .build()
     }
 
+    fun okHttpClientForApollo(token: String): OkHttpClient {
```
Comment (tamaskozmer) [inline automation/dataseedingapi/src/main/kotlin/com/instructure/dataseeding/util/CanvasRestAdapter.kt:72]: This can be private when you move the Apollo client to this object.

**Review by tamaskozmer — CHANGES_REQUESTED** [review]

File: apps/teacher/src/androidTest/java/com/instructure/teacher/ui/pages/SpeedGraderPage.kt (hunk 3 above)
Comment (kdeakinstructure) [reply apps/teacher/src/androidTest/java/com/instructure/teacher/ui/pages/SpeedGraderPage.kt:82]: This came after merging master into my branch.

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by tamaskozmer — APPROVED** [review]
